akka-cluster-typed/src/main/scala/akka/cluster/typed/internal/receptionist/ClusterReceptionist.scala

```diff
@@ -26,43 +27,49 @@ private[typed] object ClusterReceptionist extends ReceptionistBehaviorProvider {
   type SubscriptionsKV[K <: AbstractServiceKey] = ActorRef[ReceptionistMessages.Listing[K#Protocol]]
   type SubscriptionRegistry = TypedMultiMap[AbstractServiceKey, SubscriptionsKV]
 
-  private final val ReceptionistKey = ORMultiMapKey[ServiceKey[_], ActorRef[_]]("ReceptionistKey")
-  private final val EmptyORMultiMap = ORMultiMap.empty[ServiceKey[_], ActorRef[_]]
+  private final val ReceptionistKey = ORMultiMapKey[ServiceKey[_], Entry]("ReceptionistKey")
+  private final val EmptyORMultiMap = ORMultiMap.empty[ServiceKey[_], Entry]
 
-  case class ServiceRegistry(map: ORMultiMap[ServiceKey[_], ActorRef[_]]) extends AnyVal {
+  case class Entry(ref: ActorRef[_], systemUid: Long) {
+    val uniqueAddress: UniqueAddress = UniqueAddress(ref.path.address, systemUid)
```

Comment:
Note that host/port is None for local addresses, and that might not be what you want in the Entry.
For `ref.path.hasLocalScope` you should use `cluster.selfAddress`